lib/mayaUsd/ufe/UsdConnectionHandler.cpp

```diff
@@ -65,6 +65,43 @@ bool isConnected(const PXR_NS::UsdAttribute& srcUsdAttr, const PXR_NS::UsdAttrib
     return false;
 }
 
+PXR_NS::SdrShaderNodeConstPtr
+_GetShaderNodeDef(const PXR_NS::UsdPrim& prim, const PXR_NS::TfToken& attrName)
+{
+    UsdPrim           targetPrim = prim;
```

Comment:
Traverse into layers of NodeGraphs until a shader is found. Similar to what we do when [traversing node graphs](https://github.com/Autodesk/maya-usd/blob/dev/lib/mayaUsd/fileio/shading/shadingModeUseRegistry.cpp#L669) in the material importer.